### PR TITLE
Added Managed Disks sample

### PR DESCRIPTION
**Pull request by jamesbannan**
Added sample-managed-disks folder which contains a deployment template for deploying the same demo environment contained within azureDeploy.json, but using Managed Disk resources instead of traditional Storage Accounts and VM disks.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jamesbannan__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        "description": "The administrative user on the Windows VM."
+      }
+    },
+    "vmAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The administrative user password."
+      }
+    },
+    "vNetName": {
```
Comment (bmoore-msft) [inline storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json]: move to variable

File: storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    "diskCreateOption": "empty",
+    "diskAccountType": [ "Standard_LRS", "Standard_LRS", "Standard_LRS", "Premium_LRS" ],
+    "diskSizeGB": [ "100", "100", "100", "1023" ],
+    "vmImagePublisher": "MicrosoftWindowsServer",
+    "vmImageOffer": "WindowsServer",
+    "vmOSDiskName": "[concat(variables('vmName'),'-','OSDisk')]",
+    "vmSize": "Standard_DS3_v2",
+    "vmName": "demo-01",
+    "vmWindowsOSVersion": "2016-Datacenter",
+    "vmSubnetRef": "[concat(resourceId('Microsoft.Network/virtualNetworks', parameters('vNetName')), '/subnets/', variables('vNetSubnet1Name'))]",
```
Comment (bmoore-msft) [inline storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json]: just use resourceId() here, a little easier to read (no nesting)

File: storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json (hunk 1 above)
Comment (jamesbannan) [inline storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json]: Done.

File: storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json (hunk 2 above)
Comment (jamesbannan) [inline storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json]: Had to add in a new vNetId variable first using resourceId(), and then the concat() function for the vmSubnetRef variable. I've found that trying to use resourceId() for a subnet throws an error - you need to concat() it with the resource ID of the vNet first

Comment (jamesbannan) [inline storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.json]: OK - made the change

File: storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.parameters.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmAdminUserName": {
+      "value": "vm-admin"
+    },
+    "vmAdminPassword": {
+      "value": "GEN-PASSWORD"
+    },
+    "vNetName": {
```
Comment (bmoore-msft) [inline storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.parameters.json]: can remove this from the param file (surprised it didn't fail during CI)

Comment (jamesbannan) [inline storage-iops-latency-throughput-demo/sample-managed-disks/managedDisksDemo.parameters.json]: Done.

**Pull request merged**